src/test/java/bio/terra/stairway/FlightMapTest.java

```diff
@@ -0,0 +1,114 @@
+package bio.terra.stairway;
+
+import static bio.terra.stairway.StairwayMapper.getObjectMapper;
+
+import bio.terra.stairway.fixtures.FlightsTestNonPojo;
+import bio.terra.stairway.fixtures.FlightsTestPojo;
+import java.util.UUID;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Tag("unit")
+public class FlightMapTest {
+
+  private static final Logger logger = LoggerFactory.getLogger("FlightMapTest");
+
+  private static <T> void testObjectContainer(
+      T in, FlightParameterSerializer serializer, FlightParameterDeserializer<T> deserializer) {
+    FlightMap.ObjectContainer ocSer = new FlightMap.ObjectContainer(in, serializer);
+    String data = ocSer.getSerializedObjectState();
+    logger.info("Serialized Data: '{}'", data);
+
+    FlightMap.ObjectContainer ocDes = new FlightMap.ObjectContainer(data);
+    T out = ocDes.getObject(deserializer);
+    Assertions.assertEquals(in, out);
+  }
+
+  private static <T> void testObjectContainer(T in, Class<T> type) {
+    FlightParameterSerializer serializer = new DefaultFlightParameterSerializer(getObjectMapper());
+    FlightParameterDeserializer<T> deserializer =
+        new DefaultFlightParameterDeserializer<>(type, getObjectMapper());
+    testObjectContainer(in, serializer, deserializer);
+  }
+
+  @Test
+  public void objectContainers() {
+    FlightsTestPojo inPojo = new FlightsTestPojo().anint(99).astring("mystring");
+    testObjectContainer(inPojo, FlightsTestPojo.class);
+
+    FlightsTestNonPojo inNonPojo = new FlightsTestNonPojo(1.99f);
+    testObjectContainer(
+        inNonPojo, FlightsTestNonPojo.serializer(), FlightsTestNonPojo.deserializer());
+
+    Integer inInt = 100;
+    testObjectContainer(inInt, Integer.class);
+  }
+
+  @Test
+  public void toAndFromJson() {
+    FlightMap outMap = new FlightMap();
+
+    FlightsTestPojo pojoIn = new FlightsTestPojo().anint(99).astring("mystring");
+    outMap.put("mypojo", pojoIn);
+
+    FlightsTestNonPojo nonPojoIn = new FlightsTestNonPojo(1.25f);
+    outMap.put("mynonpojo", nonPojoIn, FlightsTestNonPojo.serializer());
+
+    Integer intIn = 3;
+    outMap.put("myint", intIn);
+
+    String json = outMap.toJson();
+    logger.info("JSON: '{}'", json);
+
```

Comment:
Prettified version of the JSON created by the call to `outMap.toJson()`:

```
[
    "java.util.HashMap",
    {
        "myint": "3",
        "mynonpojo": "{\"version\":2,\"uuid\":\"a5f68d4d-eaa7-4055-bbe1-882428f4ab4a\",\"value\":1.25}",
        "mypojo": "[\"bio.terra.stairway.fixtures.FlightsTestPojo\",{\"astring\":\"mystring\",\"anint\":99}]"
    }
]

```
Note that the `put()` calls for `pojoIn` and `intIn` use default API, where call for `nonPojoIn` calls new custom serializer overload.